assets/pages/regular-contributions/components/formFields.jsx

```diff
@@ -11,62 +11,89 @@ import SelectInput from 'components/selectInput/selectInput';
 import type { IsoCountry, UsState, CaState } from 'helpers/internationalisation/country';
 import type { SelectOption } from 'components/selectInput/selectInput';
 import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
-import EmailFormFieldContainer from './emailFormFieldContainer';
+import { type UserFormFieldAttribute, shouldShowError } from 'helpers/checkoutForm/checkoutForm';
+import {
+  type Action as CheckoutAction,
+  setFirstNameShouldValidate,
+  setLastNameShouldValidate,
+  setEmailShouldValidate,
+} from './contributionsCheckoutContainer/checkoutFormActions';
+import { type PageState as State } from '../regularContributionsReducer';
+import { formFields } from '../helpers/checkoutFormFields';
+
 
 // ----- Types ----- //
 
 type PropTypes = {
-  firstNameUpdate: (name: string) => void,
-  lastNameUpdate: (name: string) => void,
   stateUpdate: (value: UsState | CaState) => void,
   countryUpdate: (value: string) => void,
-  firstName: string,
-  lastName: string,
+  firstName: UserFormFieldAttribute,
+  lastName: UserFormFieldAttribute,
+  email: UserFormFieldAttribute,
+  setFirstName: (string) => void,
+  setLastName: (string) => void,
+  setEmail: (string) => void,
+  setFirstNameShouldValidate: () => void,
+  setLastNameShouldValidate: () => void,
+  setEmailShouldValidate: () => void,
   countryGroup: CountryGroupId,
   country: IsoCountry,
 };
 
 // ----- Map State/Props ----- //
 
-function mapStateToProps(state) {
+function mapStateToProps(state: State) {
+
+  const { firstName, lastName, email } = formFields(state);
 
   return {
-    firstName: state.page.user.firstName,
-    lastName: state.page.user.lastName,
+    firstName,
+    lastName,
+    email,
     countryGroup: state.common.internationalisation.countryGroupId,
     country: state.common.internationalisation.countryId,
   };
 
 }
 
-function mapDispatchToProps(dispatch: Dispatch<UserAction | PageAction>) {
+function mapDispatchToProps(dispatch: Dispatch<UserAction | PageAction | CheckoutAction>) {
 
   return {
-    firstNameUpdate: (name: string) => {
-      dispatch(setFirstName(name));
-    },
-    lastNameUpdate: (name: string) => {
-      dispatch(setLastName(name));
-    },
     stateUpdate: (value: UsState | CaState) => {
       dispatch(setStateField(value));
     },
     countryUpdate: (value: IsoCountry) => {
       dispatch(setCountry(value));
     },
+    setFirstNameShouldValidate: () => {
+      dispatch(setFirstNameShouldValidate());
+    },
+    setFirstName: (firstName: string) => {
+      dispatch(setFirstName(firstName));
+    },
+    setLastNameShouldValidate: () => {
+      dispatch(setLastNameShouldValidate());
+    },
+    setLastName: (lastName: string) => {
+      dispatch(setLastName(lastName));
+    },
+    setEmailShouldValidate: () => {
+      dispatch(setEmailShouldValidate());
+    },
+    setEmail: (email: string) => {
+      dispatch(setEmail(email));
+    },
```

Comment:
Could reduce boilerplate by making object, as discussed before for one-off, up to you.